### DISCUSSION TITLE
Any way to use msw with Sveltekit form actions?

### DISCUSSION THREAD

**Discussion by CROpie** [mswjs/msw#1669] [Q&A]
I was wondering whether anyone had investigated how to use msw intercept a sveltekit form action (https://kit.svelte.dev/docs/form-actions). The path should be http://localhost:5173/formaction?/test, but I am getting this error: 

[MSW] Error: captured a request without a matching request handler:
  • POST http://localhost:3000/undefined

Additionally, even if the request is intercepted, I'm not sure where ctx.json() in the response would end up, since the return value of the response to the fetch request seems to be handled different than in the normal method?

Any help or advice would be appreciated - especially if it's just to let me know to not spend any more time on it ;)

**Comment by kettanaito** [comment]
Hi, @CROpie! 

There's an official [Svelte usage example](https://github.com/mswjs/examples-new/tree/main/examples/with-svelte) we prepared for the upcoming release of MSW (see #1464 for the instructions on how to install and migrate to it). Please give it a try and let me know if it works with form actions.

**Reply by CROpie** [reply]
Thanks for the suggestion, I'll let you know if I can get it working. I was under the impression it was just for the load functions, but we'll see!  For some reason though at the moment I'm forced to include http://localhost:5173 in my regular fetch requests and msw because otherwise it defaults to http://localhost:3000 ? Do you know why this may be the case- I might not have set things up properly to begin with? ** I should note that this is using vitest and is for integration testing...

**Reply by kettanaito** [reply]
Not sure what that occurs but it must be on your end. MSW doesn't have any defaults for request hosts. It may be your test environment interfering, if you're running MSW in tests (e.g. `window.location` points to `localhost:3000`). 

MSW should work with form actions as well. It does for Remix but I haven't got to try Svelte. I would love to learn from you and update the Svelte example, if necessary. Thanks for looking into this.

**Comment by xerosanyam** [comment]
hey @CROpie did u find a clean way to do it?
